service/src/main/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/S3StorageConnector.java

```diff
@@ -0,0 +1,63 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.S3Object;
+
+public class S3StorageConnector implements StorageConnector {
+
+    private final AmazonS3 s3client;
+    private final String bucketName;
+
+    protected S3StorageConnector(StorageConnectorConfiguration configuration) {
+        this.bucketName = configuration.getContainerName();
+        AWSCredentials credentials = new BasicAWSCredentials(
+            configuration.getS3AccessKey(),
+            configuration.getS3SecretKey()
+        );
+        this.s3client = AmazonS3ClientBuilder
+            .standard()
+            .withCredentials(new AWSStaticCredentialsProvider(credentials))
+            .withRegion(Regions.EU_WEST_2)
```

Comment:
Same with region, we won't always be in EU_WEST_2 - omit and it let it be ready from the local environment. You may need to define these locally or setup the aws cli for it to work from your local machine though.